packages/dara-components/js/common/table/table.tsx

```diff
@@ -346,23 +364,67 @@ function Table(props: TableProps): JSX.Element {
     const [selectedRowIndices, setSelectedRowIndices] = useVariable(props.selected_indices);
 
     const [columnsProp] = useVariable(props.columns);
-    const [resolvedColumns, setResolvedColumns] = useState<ColumnProps[]>(null);
+    const [resolvedColumns, setResolvedColumns] = useState<ColumnProps[]>(() =>
+        isArray(columnsProp) ? getColumnProps(columnsProp) : null
+    );
 
+    // Resolve columns from data
     useEffect(() => {
-        // prop provided, just parse the columns provided
-        if (columnsProp) {
-            setResolvedColumns(getColumnProps(columnsProp));
-            return;
-        }
-        // prop not provided, find columns from data
-        getData(null, {
-            limit: 1,
-            offset: 0,
-        })
+        getData(
+            null,
+            {
+                limit: 1,
+                offset: 0,
+            },
+            { schema: true }
+        )
             .then((dataset) => {
                 const columns = Object.keys(dataset.data[0]);
-                const columnsWithoutIndex = columns.filter((col) => col !== '__index__');
-                setResolvedColumns(getColumnProps(columnsWithoutIndex));
+                const fieldTypes = Object.fromEntries(
+                    dataset.schema.fields.flatMap((field) => {
+                        const key = Array.isArray(field.name) ? field.name.join('_') : field.name;
+                        return [[key, { type: field.type }]];
+                    })
+                );
+
+                const columnsWithoutGeneratedIndex = columns.filter((col) => col !== INDEX_COL);
+                let processsedColumns: ColumnProps[];
+                if (columnsProp) {
+                    // Prop provided, parse the columns provided and map them to the columns from data
+                    // Limitation: If there are columns with duplicate names, data from only one of them will be shown
+                    const reverseColumnIdMap = Object.fromEntries(
+                        columnsWithoutGeneratedIndex.map((col) => [
+                            extractColumnLabel(col, col.startsWith('__index__')),
```

Comment:
I think we have a constant defined for the `__index__` prefix, we should probably use it everywhere instead of the magic string?